apis/apiextensions/v1/composition_revision_types.go

```diff
@@ -38,7 +38,9 @@ const (
 type CompositionRevisionSpec struct {
 	// CompositeTypeRef specifies the type of composite resource that this
 	// composition is compatible with.
-	// +immutable
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:XValidation:rule="!has(oldSelf.value) || has(self.value)", message="Value is required once set"
```

Comment:
Does this actually make the field immutable? It looks like it prevents it from being removed, but doesn't prevent it from being changed.

Reply:
@negz Actually I have already added `+kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"` just below that line which will look into making the field immutable.

Reply:
Do we need both rules, `self == oldSelf` and also `!has(oldSelf.value) || has(self.value)`?

Also, does the latter rule actually prevent a field being removed? The logic appears to be "if old self doesn't have the value, or self has the value". Should it be `&&` not `||`?

Reply:
@negz I have removed that rule now and we only check `self==oldself`